Add new DA tutorial and docs (#131)
rollkit/docs: blog/2023-03-5-sovereign-rollups-on-bitcoin-with-rollkit.mdx

---
slug: sovereign-rollups-on-bitcoin
title: "Sovereign rollups on Bitcoin with Rollkit"
authors: [rollkit]
tags: [Rollkit, Bitcoin]

> description: Last week, we introduced Rollkit, a modular framework for rollups. Today, we are proud to announce that Rollkit is the first rollup framework to support sovereign rollups on Bitcoin. A module that allows Rollkit rollups to use Bitcoin for data availability is now available. The module is in alpha, but we invite curious developers to experiment with it.

Reviewer: ```suggestion
description: Last week, we introduced Rollkit, a modular framework for rollups. Today, we are proud to announce that Rollkit is the first rollup framework to support sovereign rollups on Bitcoin. This allows Rollkit rollups to use Bitcoin for data availability. The implementation is in alpha, but we invite curious developers to experiment with it.
```

---

![rollkit-bitcoin](../static/img/bitcoin-rollkit/rollkit-bitcoin.png)

> Last week, we introduced Rollkit, a modular framework for rollups. Today, we are proud to announce that Rollkit is the first rollup framework to support sovereign rollups on Bitcoin. An early research implementation of a module that allows Rollkit rollups to use Bitcoin for data availability is now available.

Reviewer: ```suggestion
Last week, we introduced Rollkit, a modular framework for rollups. Today, we are proud to announce that Rollkit is the first rollup framework to support sovereign rollups on Bitcoin. An early research implementation allows Rollkit rollups to use Bitcoin for data availability.
```

Before the introduction of Taproot, data that could be inscribed in a Bitcoin single transaction was limited to 80 bytes. The Taproot upgrade increased the amount of data a transaction could inscribe, from 80 bytes to a full block’s worth of data (around 4MB). In short, it became easier to post big blobs of data to the Bitcoin blockchain.

Since then, the usage of Ordinals for NFT inscriptions and Taproot utilization has [kicked off](https://dune.com/dataalways/ordinals). Eric Wall found that at the time of [his tweet](https://twitter.com/ercwl/status/1619671451417862145), posting data on Bitcoin was 7x cheaper than Ethereum. Now that there are thousands of inscriptions on Bitcoin, it has become clear that sovereign rollups and an ecosystem of dapps on Bitcoin could become a reality. The missing piece: a rollup framework for easily integrating Bitcoin as a data availability layer.

## Integrating Bitcoin as a data availability layer into Rollkit

> Rollkit is a modular framework for rollups, where developers can plug-in custom execution layers and data availability layers. Initially, Rollkit only supported Celestia as an option for data availability and consensus. Now, Bitcoin is an option, thanks to an early research implementation of a Bitcoin data availability module for Rollkit. The new module makes it possible to deploy sovereign rollups on Bitcoin. In this case, sovereign rollups manage their own execution and settlement while offloading consensus and data availability to Bitcoin.

Reviewer: ```suggestion
Rollkit is a modular framework for rollups, where developers can plug-in custom execution layers and data availability layers. Initially, Rollkit only supported Celestia as an option for data availability and consensus. Now, Bitcoin is an option, thanks to an early research implementation of a Bitcoin data availability module for Rollkit. In this case, sovereign rollups manage their own execution and settlement while offloading consensus and data availability to Bitcoin.
```
